### PR TITLE
Added improved error handling, more consistent comments, and improved the lookup of most recent transaction by adding a DateTime column to transaction

### PR DESCRIPTION
**Pull request by carsonwolber**
In this PR I address the issues that have been presented preventing past PRs from getting approval. Notably this includes
– adding a try-catch block when initially dealing with data fetched from Plaid so we're better prepared in the event Plaid doesn't retrurn in the expected format
– improved client-side handling of fetch requests to log the error and the function if a nework error occurs
– removed a console.log statement and instead added error handling
– Added a "time" column to the transactions database so we can more efficiently and reliably display the most recently fetched transaction data

In addition to this, I added a few other small quality of life improvements I noticed while implementing the above features: 
– I changed some of the Ag-grid configuration to now show the number of associated-rows as well as having the indented rows collapse by default
– I removed unnecessary comments and improved my comment consistency when I noticed capitalization was inconsistent across files. 

Screenshot 2024-07-09 at 11 43 40 AM
Screenshot 2024-07-09 at 11 35 24 AM

### CODE REVIEW

**Comment by William-La** [comment]
Discussed offline, but going forward let's respond to comments and suggestions in the same PR as they are made. We can update existing PRs by committing and pushing to the branch the PR is created from.

File: backend/data_handling/data_processing.py [hunk 1]
```diff
@@ -10,19 +10,26 @@
...
+                        "amount": round(random.uniform(0.01, 5000), 2),
+                        "date": transaction["date"],
+                        "currency": transaction["iso_currency_code"],
+                        "detailed": transaction["personal_finance_category"]["detailed"],
+                        "primary": transaction["personal_finance_category"]["primary"]
+                    }
+                    cleaned_data.append(clean_transaction)
+                return cleaned_data
+    except Exception as e:
+        (f"Error cleaning transaction data: {e}")
```
Comment (William-La) [inline backend/data_handling/data_processing.py]: Currently this line evaluates the f string but does not do anything with it-- we should throw/log the error prior to returning

**Review by William-La — APPROVED** [review]
The changes look good-- thanks for the improvements. Address the one error raising comment before landing.

File: backend/data_handling/data_processing.py (hunk 1 above)
Comment (carsonwolber) [reply backend/data_handling/data_processing.py]: Just fixed this in commit a365695

**Pull request merged**